services/repository/files/update.go

```diff
@@ -448,6 +448,10 @@ func CreateOrUpdateFile(ctx context.Context, t *TemporaryUploadRepository, file
 			return err
 		}
 		if !exist {
+			_, err := file.ContentReader.(io.Seeker).Seek(0, io.SeekStart)
```

Comment:
Good catch! I think the type of `ContentReader` should be changed to `io.ReadSeeker`, then the dynamic type casting could be avoided.

Reply:
Thanks for your review! I have updated the code. Would you mind take a look again?

Reply:
Erm… I think what wxiaoguang meant is that `file.ContentReader` should **always** be a `io.ReadSeeker` so that the cast here is unnecessary

Reply:
@delvh Sorry, i think I misunderstood that. I've changed the type of `file.ContentReader` to `io.ReadSeeker`.

Reply:
The new commit looks good to me. Luckily, all values assigned to `file.ContentReader` is `io.ReadSeeker`, so no more work need to do 😁